### ISSUE TITLE
[EVENT REQUEST] Table 27 - OnModify - OnBeforePlanningAssignmentItemChange

### ISSUE THREAD

**Issue by Denushaponov** [microsoft/ALAppExtensions#22512]
Please add the following event and IsHandled to the OnModify trigger for table 27
```
  trigger OnModify()
    var
        IsHandled: Boolean;
    begin
        UpdateReferencedIds();
        SetLastDateTimeModified();
      
        OnBeforePlanningAssignmentItemChange(Rec, xRec, PlanningAssignment, IsHandled);
	if not IsHandled then
        PlanningAssignment.ItemChange(Rec, xRec);

        UpdateItemUnitGroup();
    end;

    [IntegrationEvent(false, false)]
    local procedure OnBeforePlanningAssignmentItemChange(var Item: Record Item; xItem: Record Item; PlanningAssignment: Record "Planning Assignment"; var IsHandled: Boolean)
    begin
    end;

```
Internal work item: [AB#536757](https://dynamicssmb2.visualstudio.com/1fcb79e7-ab07-432a-a3c6-6cf5a88ba4a5/_workitems/edit/536757)

**Comment by AlexanderYakunin** [comment]
Using IsHandled here is very risky approach as you can miss important code on modify. Please explain what exactly you want to achieve by this?

**Comment by Denushaponov** [comment]
I'm Really sorry for that, I replaced the code  with the more precise version
I need to run PlanningAssignment.ItemChange(Rec, xRec); and customized ItemChange function from conditional statement

**Comment by AleksandricMarko** [comment]
Hello @Denushaponov 
Just to check is this still valid request?

**Comment by Denushaponov** [comment]
I believe so.

чт, 30 мая 2024 г., 2:34 PM Marko Aleksandric ***@***.***>:

**Comment by AleksandricMarko** [comment]
Already implemented from BC 22.5 and newer
